test/blockchaintest/blockchaintest_loader.cpp

```diff
@@ -124,7 +115,19 @@ BlockchainTest load_blockchain_test_case(const std::string& name, const json::js
     bt.rev = to_rev_schedule(j.at("network").get<std::string>());
 
     for (const auto& el : j.at("blocks"))
-        bt.test_blocks.emplace_back(load_test_block(el, bt.rev.get_revision(0)));
+    {
+        // `rlp_decoded` holds the `FixtureBlock` element with the relevant block data for
+        // invalid blocks within a test. It should be a sibling element to `expectException`.
+        if (const auto it = el.find("rlp_decoded"); it != el.end())
+        {
+            auto test_block = load_test_block(el.at("rlp_decoded"), bt.rev.get_revision(0));
+            assert(el.find("expectException") != el.end());
+            test_block.valid = false;
```

Comment:
maybe beter
```suggestion
            assert(test_block.valid == false);
```

Reply:
not the same thing, since `rlp_decoded` won't contain `expectException`

Reply:
I meant that the call to `load_test_block()` on the previous line already always sets `test_block.valid = false` iff `expectException` is present, so these conditions are equivalent and you don't need to do second assignment, just assert that it was already set.

But maybe you plan to change this logic anyway.

Reply:
> on the previous line already always sets test_block.valid = false iff expectException is present

yes, this is the catch! `expectException` is never present in `rlp_decoded`, its only present in the `blocks` array element of type `InvalidFixtureBlock`. And yes, I think I'll need to change this logic somehow, but the eofwrap stuff is getting in the way :/. 

So to illustrate, valid blocks are like
```
blocks: [{decoded fields like header, txs etc, but never expectException}[
```
invalid ones:
```
blocks: [{rlp:..., expectException: ..., rlp_decoded: {decoded fields like header, txs etc, but never expectException}}]
```

So as stated above, code works correctly, but for the wrong reasons